src/aics-image-viewer/components/App/index.tsx

```diff
@@ -243,43 +259,58 @@ const App: React.FC<AppProps> = (props) => {
     (image: Volume, channelIndex: number, isInitialLoad: boolean): void => {
-      // TODO this was once a search by name - is that still necessary or will the index always be correct?
-      const thisChannelSettings = channelSettings[channelIndex];
-      const viewerState = useViewerState.getState();
-      const { channelsToResetOnLoad, useDefaultViewerChannelSettings } = viewerState;
+      // TODO this was once a search by name - is that still necessary or will
+      // the index always be correct?
+      const { channelsToResetOnLoad, useDefaultViewerChannelSettings, channelSettings } = useViewerState.getState();
+      const channelState = channelSettings[channelIndex];
+
       const viewerChannelSettings = useDefaultViewerChannelSettings
         ? getDefaultViewerChannelSettings()
         : props.viewerChannelSettings;
       const thisChannel = image.getChannel(channelIndex);
-      const noLut = !thisChannelSettings || !thisChannelSettings.controlPoints || !thisChannelSettings.ramp;
+      const noLut = !channelState || !channelState.controlPoints || !channelState.ramp;
 
-      if (isInitialLoad || noLut || channelsToResetOnLoad.includes(channelIndex)) {
+      const hasOldRange = channelRangesRef.current[channelIndex] !== undefined;
+      const initializeToExistingRange = channelState && channelState.keepIntensityOnNewVolume && hasOldRange;
+      const initializeToDefaults = isInitialLoad && !initializeToExistingRange;
+
+      if (initializeToDefaults || noLut || channelsToResetOnLoad.includes(channelIndex)) {
         // This channel needs its LUT initialized
         const { ramp, controlPoints } = initializeLut(image, channelIndex, viewerChannelSettings);
 
+        // Initialize isovalue to channel settings or default to midpoint of data range
+        const name = image.channelNames[channelIndex];
+        const initSettings = viewerChannelSettings && findFirstChannelMatch(name, channelIndex, viewerChannelSettings);
+        const defaultIsovalue = thisChannel.rawMin + (thisChannel.rawMax - thisChannel.rawMin) / 2;
+        const isovalue = initSettings?.isovalue ?? defaultIsovalue;
+
         changeChannelSetting(channelIndex, {
           controlPoints: controlPoints,
           ramp: controlPointsToRamp(ramp),
           // set the default range of the transfer function editor to cover the full range of the data type
           plotMin: thisChannel.rawMin,
           plotMax: thisChannel.rawMax,
-          isovalue: thisChannel.rawMin + (thisChannel.rawMax - thisChannel.rawMin) / 2,
+          isovalue,
+        });
+      } else {
+        // Expand the plot min and max to include the current data range as
+        // needed. This keeps the domain visually consistent when replaying
+        // through time or Z slices.
+        //
+        // NOTE: This must use the most up-to-date channel plot range, because
+        // onChannelLoaded can be called multiple times per channel during
+        // loading.
```

Comment:
(see above where we're getting channelSettings out of state)

Reply:
It's getting passed as a generic callback into `sceneLoader.createVolume()`, but is also being used as a callback for specific loading functionality (`view3d.setTime` and `sceneLoader.loadScene`).